Kafka exception not suppressed on Job cancel

I saw this once or twice, Jet 0.6, closing a job with a Kafka source

```
12:19:52.854 WARN  hz._hzInstance_1_amsjug.jet.blocking.thread-3 c.h.j.i.e.TaskletExecutionService - [127.0.0.1]:5701 [xxx] [0.6] Exception in ProcessorTasklet{Kafka Source#3} 
org.apache.kafka.common.errors.InterruptException: java.lang.InterruptedException
    at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.maybeThrowInterruptException(ConsumerNetworkClient.java:447)
    at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:254)
    at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:214)
    at org.apache.kafka.clients.consumer.internals.ConsumerNetworkClient.poll(ConsumerNetworkClient.java:190)
    at org.apache.kafka.clients.consumer.internals.Fetcher.getTopicMetadata(Fetcher.java:314)
    at org.apache.kafka.clients.consumer.KafkaConsumer.partitionsFor(KafkaConsumer.java:1481)
    at com.hazelcast.jet.kafka.impl.StreamKafkaP.assignPartitions(StreamKafkaP.java:116)
    at com.hazelcast.jet.kafka.impl.StreamKafkaP.complete(StreamKafkaP.java:165)
    at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:253)
    at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:143)
    at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:136)
    at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:187)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.InterruptedException: null
    ... 17 common frames omitted
```

@viliam-durina  Thinks it should be suppressed, not a real issue but will cause confusion

--- a/hazelcast-jet-kafka/src/main/java/com/hazelcast/jet/kafka/impl/StreamKafkaP.java
+++ b/hazelcast-jet-kafka/src/main/java/com/hazelcast/jet/kafka/impl/StreamKafkaP.java
@@ -162,27 +162,25 @@ public final class StreamKafkaP<K, V, T> extends AbstractProcessor {
 
     @Override
     public boolean complete() {
-        assignPartitions(true);
-
         if (!emitFromTraverser(traverser)) {
             return false;
         }
 
         ConsumerRecords<K, V> records = null;
-        if (!currentAssignment.isEmpty()) {
-            try {
+        try {
+            assignPartitions(true);
+            if (!currentAssignment.isEmpty()) {
                 records = consumer.poll(POLL_TIMEOUT_MS);
-            } catch (InterruptException e) {
-                // note this is Kafka's exception, not Java's
-                Thread.currentThread().interrupt();
-                return false;
             }
+        } catch (InterruptException e) {
+            // note this is Kafka's exception, not Java's
+            Thread.currentThread().interrupt();
+            return false;
         }
 
-        traverser = isEmpty(records) ?
-                watermarkSourceUtil.handleNoEvent()
-                :
-                traverseIterable(records).flatMap(record -> {
+        traverser = isEmpty(records)
+                ? watermarkSourceUtil.handleNoEvent()
+                : traverseIterable(records).flatMap(record -> {
                     offsets.get(record.topic())[record.partition()] = record.offset();
                     T projectedRecord = projectionFn.apply(record);
                     if (projectedRecord == null) {

